src/metal.cs

```diff
@@ -1741,6 +2001,69 @@ partial interface MTLRenderCommandEncoder : MTLCommandEncoder {
 #endif
 		[Export ("drawIndexedPatches:patchIndexBuffer:patchIndexBufferOffset:controlPointIndexBuffer:controlPointIndexBufferOffset:indirectBuffer:indirectBufferOffset:")]
 		void DrawIndexedPatches (nuint numberOfPatchControlPoints, [NullAllowed] IMTLBuffer patchIndexBuffer, nuint patchIndexBufferOffset, IMTLBuffer controlPointIndexBuffer, nuint controlPointIndexBufferOffset, IMTLBuffer indirectBuffer, nuint indirectBufferOffset);
+
+		[Mac (10,13), NoiOS, NoTV, NoWatch]
+#if XAMCORE_4_0
+		[Abstract]
+#endif
+		[Export ("setViewports:count:")]
+		void SetViewports (IntPtr viewports, nuint count);
```

Comment:
Yes, the generator thinks managed arrays maps to NSArray on the native side 😒. And since this is a protocol, binding it manually won't work as expected. As above, it might be possible to use unsafe code `(MTLViewPort* viewports, nuint count)`.

This would be another good candidate for BindAs.